### PR TITLE
New Public/Private Dashboard

### PR DESCRIPTION
**Pull request by alexsporn**
This PR adds the new dashboard with login support.
Basic auth was replaced with a login mask and JWT authorization.
This PR also secures the exposed API endpoints and allows read-only access to the dashboard API if not using a JWT token.
The default configuration is a zeroed out hashed password and salt to avoid having default passwords deployed everywhere.

The included dashboard version includes all the updated indexation, address type and hashing changes.

### CODE REVIEW

**Comment by alexsporn** [comment]
This closes #787

File: plugins/dashboard/routes.go [hunk 1]
```diff
@@ -23,90 +21,141 @@ const (
...
+				return false
+			}
 		}
-		return e.Blob(http.StatusOK, contentType, data)
+
+		return true
 	}
-	staticBlob, err := appBox.Find(e.Request().URL.Path)
+
+	apiBindAddr := deps.NodeConfig.String(restapi.CfgRestAPIBindAddress)
```
Comment (muXxer) [inline plugins/dashboard/routes.go:121]: maybe split the port and use localhost internally?

File: plugins/dashboard/jwt.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	return nil
+}
+
+func (c *jwtClaims) compare(field string, expected string) bool {
+	if field == "" {
+		return false
+	}
+	if subtle.ConstantTimeCompare([]byte(field), []byte(expected)) != 0 {
+		return true
+	} else {
```
Comment (muXxer) [inline plugins/dashboard/jwt.go]: you can remove the `else`

File: plugins/dashboard/routes.go [hunk 3]
```diff
@@ -23,90 +21,141 @@ const (
...
+	notFoundMiddleware := func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) (err error) {
+			return ErrNotFound
+		}
+	}
+
+	return []echo.MiddlewareFunc{
+		jwtAuth.Middleware(jwtAuthSkipper),
+		middleware.ProxyWithConfig(config),
+		notFoundMiddleware,
```
Comment (muXxer) [inline plugins/dashboard/routes.go:162]: Do we need this middleware?

File: plugins/dashboard/routes.go [hunk 4]
```diff
@@ -139,176 +188,53 @@ func enforceMaxOneDotPerURL(next echo.HandlerFunc) echo.HandlerFunc {
...
-		}
-	}()
 
-	// this function sends the initial values for some topics
-	sendInitValue := func(client *websockethub.Client, initValuesSent map[byte]struct{}, topic byte) {
-		if _, sent := initValuesSent[topic]; sent {
-			return
+	if len(request.JWT) > 0 {
+		// Verify JWT is still valid
+		if !jwtAuth.VerifyJWT(request.JWT) {
```
Comment (muXxer) [inline plugins/dashboard/routes.go:213]: Can we do an early return if the auth is still correct, or do we need to issue a new one?

```go
	return c.JSON(http.StatusOK, map[string]string{
		"jwt": request.JWT,
	})``

**Pull request merged**